### PR TITLE
all required changes - added logger, configured default Makefile and …

### PR DESCRIPTION
**Pull request by eMaerthin**
…additionally added example how to run validation

### CODE REVIEW

File: examples/detector.c [hunk 1]
```diff
@@ -575,9 +576,15 @@ void test_detector(char *datacfg, char *cfgfile, char *weightfile, char *filenam
     double time;
     char buff[256];
     char *input = buff;
+    char buff2[256];
+    char *output2 = buff2;
```
Comment (michalc-mit) [inline examples/detector.c]: please add comment or use more elaborate naming

File: examples/detector.c [hunk 2]
```diff
@@ -575,9 +576,15 @@ void test_detector(char *datacfg, char *cfgfile, char *weightfile, char *filenam
...
     int j;
     float nms=.3;
+    int iter = 0;
     while(1){
+        ++iter;
+        FILE* f;
+        f = fopen(detector_log,"a");
```
Comment (michalc-mit) [inline examples/detector.c]: check if descriptor is valid

File: examples/detector.c [hunk 3]
```diff
@@ -612,19 +620,42 @@ void test_detector(char *datacfg, char *cfgfile, char *weightfile, char *filenam
...
+            }
+        }
+        fclose(f);
+        if(output2){
+            save_image(im, output2);
+        }
+        else if(outfile){
             save_image(im, outfile);
         }
         else{
```
Comment (michalc-mit) [inline examples/detector.c]: formatting

File: examples/detector.c (hunk 2 above)
Comment (eMaerthin) [reply examples/detector.c]: done below

**Pull request merged**